Makefile.am

```diff
@@ -158,14 +158,66 @@ base_files = \
 bin_PROGRAMS += csd
 
 csd_SOURCES = \
-	${base_files} \
+	${BASE_FILES} \
 	src/main.cpp
 
 csd_LDADD = $(GLIB_LIBS) $(AVAHI_LIBS) $(GST_LIBS)
 
-.PHONY: samples
-samples:
-	@$(MAKE) -C samples
+#Samples
+EXTRA_PROGRAMS = samples/camera-sample-custom samples/camera-sample-client samples/camera-sample-mavlink-client
+
+if HAVE_REALSENSE
+EXTRA_PROGRAMS += samples/camera-sample-realsense
+endif
+
+samples_camera_sample_custom_LDADD = ${csd_LDADD}
+samples_camera_sample_realsense_LDADD = ${csd_LDADD}
+samples_camera_sample_client_LDADD = $(GLIB_LIBS) $(AVAHI_LIBS)
+samples_camera_sample_mavlink_client_LDADD = $(GLIB_LIBS) $(AVAHI_LIBS)
+
+samples_camera_sample_custom_SOURCES = \
+	$(BASE_FILES) \
+	src/main.cpp \
+	samples/stream_builder_custom.h \
+	samples/stream_builder_custom.cpp \
+	samples/stream_custom.cpp \
+	samples/stream_custom.h
+
+samples_camera_sample_realsense_SOURCES = \
+	$(BASE_FILES) \
+	src/main.cpp \
+	samples/stream_builder_realsense.h \
+	samples/stream_builder_realsense.cpp \
+	samples/stream_realsense.cpp \
+	samples/stream_realsense.h
+
+samples_camera_sample_client_SOURCES = \
+	samples/main_sample_client.cpp \
+	src/log.cpp \
+	src/log.h \
+	src/glib_mainloop.cpp \
+	src/glib_mainloop.h \
+	src/mainloop.cpp \
+	src/mainloop.h
+
+samples_camera_sample_mavlink_client_SOURCES = \
+	samples/main_sample_mavlink_client.cpp \
+	src/log.cpp \
+	src/log.h \
+	src/glib_mainloop.cpp \
+	src/glib_mainloop.h \
+	src/mainloop.cpp \
+	src/mainloop.h \
+	src/pollable.cpp \
+	src/pollable.h \
+	src/socket.cpp \
+	src/socket.h \
+	src/util.c \
+	src/util.h
+
+.PHONY: samples samples2
+samples: include/mavlink/ardupilotmega/mavlink.h
+	$(MAKE) $(EXTRA_PROGRAMS)
```

Comment:
samples2 was a left over. I'll fix that.
I need to generate mavlink.h before start building the samples. BUILT_SOURCES are guaranteed to be built before other targets only when make all or make dist is used. So for make samples I would need to enforce that. If I use:
```
samples: $(EXTRA_PROGRAMS)
```
It would generate mavlink.h.

If I use:
```
samples: include/mavlink/ardupilotmega/mavlink.h $(EXTRA_PROGRAMS)
```
It may fail in parallel builds.

Do you have another suggestion for that?


I